File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    answer = ''
    target = []
    
    cards_1 = deque(cards1)
    cards_2 = deque(cards2)
    
    cnt_1 = 0
    cnt_2 = 0
    
    cards_1.append('')
    cards_2.append('')
    
    for i in range(len(goal)) :
        if cards_1[cnt_1] == goal[i] :
            target.append(cards_1[cnt_1])
            cnt_1 += 1
        elif cards_2[cnt_2] == goal[i] :
            target.append(cards_2[cnt_2])
            cnt_2 += 1
        else :
            answer = "No"
            break
    
    if answer != "No" :
        if target == goal :
            answer = "Yes"
        else :
            answer ="No"
    
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;


class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        
        int idx_1 = 0;
        int idx_2 = 0;
        
        for (int i=0; i < goal.length; i++) {
            if (idx_1 < cards1.length && goal[i].equals(cards1[idx_1])) {
                idx_1 += 1;
            }
            else if (idx_2 < cards2.length && goal[i].equals(cards2[idx_2])) {
                idx_2 += 1;
            }
            else {
                return "No";
            }
        }
        
        return answer;
    }
}